src/ccnl-core/src/ccnl-relay.c

```diff
@@ -907,7 +907,7 @@ ccnl_fib_rem_entry(struct ccnl_relay_s *relay, struct ccnl_prefix_s *pfx,
             }
             ccnl_prefix_free(fwd->prefix);
             ccnl_free(fwd);
-            break;
+            return res;
```

Comment:
It looks like the dereference and use of `fwd` is more of a bug than anything else. The debug message states it "added FIB via $face", while this function aims to remove an entry from the FIB, so IMHO it makes more sense to just remove the if-block below. That the entry will be removed is already outputted in line 893.